src/base/bittorrent/torrenthandleimpl.cpp

```diff
@@ -1368,6 +1368,41 @@ void TorrentHandleImpl::renameFile(const int index, const QString &name)
     m_nativeHandle.rename_file(lt::file_index_t {index}, Utils::Fs::toNativePath(name).toStdString());
 }
 
+void TorrentHandleImpl::renameFolder(const QString &oldPath, const QString &newPath)
+{
+    if (!isValid()) return;
+    const QString sanatizedOldPath = oldPath + (oldPath.endsWith("/") ? "" : "/");
+    const QString sanatizedNewPath = newPath + (newPath.endsWith("/") ? "" : "/");
+
+    // Check for overwriting
+#if defined(Q_OS_WIN)
+    const Qt::CaseSensitivity caseSensitivity = Qt::CaseInsensitive;
+#else
+    const Qt::CaseSensitivity caseSensitivity = Qt::CaseSensitive;
+#endif
+    for (int i = 0; i < filesCount(); ++i) {
+        const QString currentPath = filePath(i);
+
+        if (currentPath.startsWith(sanatizedOldPath))
```

Comment:
Seems this checking isn't necessary.

Reply:
I'm looking into this now.

Reply:
@glassez This logic was included in the function before I took it. I'm not sure what the original author was trying to accomplish here, but I can remove it if you want.